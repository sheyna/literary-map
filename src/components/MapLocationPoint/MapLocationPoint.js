import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faLocationPin } from '@fortawesome/free-solid-svg-icons'; // alt faThumbTack
import styles from './MapLocationPoint.module.css';
import PropTypes from 'prop-types';

function MapLocationPoint(props) {
	const {location} = props;
	const {thumb, title, positionTop, positionLeft} = props.location;
	const pointStyles = {
		top: positionTop + "%",
		left: positionLeft + "%",
	}
	return (
		<div 
			className={styles.MapLocationPoint} 
			style={pointStyles}
			onClick={() => {
				props.openInfoBox(true, location);
			}}
		>
			{ thumb ? <img src={"img/" + thumb} alt={title}/> : <FontAwesomeIcon className={styles.locationPin} icon={faLocationPin} /> }
		</div>
	);
}

MapLocationPoint.propTypes = {
	location: PropTypes.shape({
		title: PropTypes.string,
		book: PropTypes.string,
		author: PropTypes.string,
		thumb: PropTypes.string,
		image: PropTypes.string,
		positionTop: PropTypes.number,
		positionLeft: PropTypes.number,
		townName: PropTypes.string,
		bodyText: PropTypes.string,
	}),
	openInfoBox: PropTypes.func,
};

export default MapLocationPoint;
