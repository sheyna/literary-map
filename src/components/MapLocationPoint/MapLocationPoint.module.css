.MapLocationPoint {
	z-index: 40;
	position: absolute;
}
.MapLocationPoint:hover {
    cursor: pointer;
}
.MapLocationPoint i {
	color: #bd4089;
	font-size: 1.4rem;
}
.MapLocationPoint i:hover {
	opacity: 0.6;
	transform: scale(1.3);
	transform-origin: bottom;
}
.MapLocationPoint img {
	width: 2rem;
	width: 100%;
}
.MapLocationPoint img:hover {
	transform: scale(1.1);
}
@media screen and (min-width: 380px) {
	.MapLocationPoint {
		width: 3rem;
	}
}
@media screen and (min-width: 1025px) {
	.MapLocationPoint img {
		width: 5rem;
	}
}