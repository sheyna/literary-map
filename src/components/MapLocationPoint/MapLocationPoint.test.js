import ReactTestRenderer from 'react-test-renderer';
import MapLocationPoint from  './MapLocationPoint';

const mockFn = jest.fn();

const testLocation = {
	"locationId": 9,
	"title": "Elizabeth Gaskell's Home",
	"book": "Mary Barton, North and South, Cranford, Wives and Daughters",
	"author": "Elizabeth Gaskell",
	"positionTop": 38,
	"positionLeft": 48,
	"townName":  "Manchester",
	"bodyText": "Gaskell's historic home was restored in 2014. It is now a museum to the author and her times."
};

describe('MapElement component', () => {
  it('should match snapshot', () => {
    const component = ReactTestRenderer.create(
      <MapLocationPoint
        location={testLocation}
        openInfoBox={mockFn}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
