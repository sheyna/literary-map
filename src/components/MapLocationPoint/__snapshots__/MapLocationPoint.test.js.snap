// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapElement component should match snapshot 1`] = `
<div
  className="MapLocationPoint"
  onClick={[Function]}
  style={
    Object {
      "left": "48%",
      "top": "38%",
    }
  }
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-location-pin locationPin"
    data-icon="location-pin"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 384 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M384 192c0 87.4-117 243-168.3 307.2c-12.3 15.3-35.1 15.3-47.4 0C117 435 0 279.4 0 192C0 86 86 0 192 0S384 86 384 192z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</div>
`;
